File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>

int spaces(string passage);

int main(void)
{
    // getting the user input
    string passage = get_string("Enter the passage here: ");
    spaces(passage);
}

// calculating and returning grade
int spaces(string passage)
{
    float letters = 0;
    float sentences = 0;
    int spaces = 0;
    // counting letters, spaces and sentences
    for (int i = 0; passage[i] != '\0'; i++)
    {
        if (isalpha(passage[i]))
            letters++;
        if (passage[i] == ' ')
            spaces++;
        else if (passage[i] == '.' || passage[i] == '?' || passage[i] == '!')
            sentences++;
    }

    // calculating index variables
    float words = spaces + 1;
    float avg_letters = (letters / words) * 100.00;
    float avg_sentences = (sentences / words) * 100.00;

    // calculating index and rounding it to nearest integer
    float index = (0.0588 * avg_letters) - (0.296 * avg_sentences) - 15.8;
    int grade = round(index);

    // returning grade
    if (grade <= 1)
        printf("Before Grade 1\n");
    else if (grade >= 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", grade);
    return 0;
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
length = len(text)
words = 1
L = 0
s = 0
for i in range(length):
    if text[i].isalnum():
        L += 1
    if text[i] == " ":
        words += 1
    elif text[i] in ['.', '?', '!']:
        s += 1
L = (L / words) * 100
s = (s/words) * 100
readability = round(0.0588 * L - 0.296 * s - 15.8)
if readability < 1:
    print("Before Grade 1")
elif readability >= 16:
    print('Grade 16+')
else:
    print(f'Grade {readability}')


# Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8,
# where L is the average number of letters per 100 words in the text,
# and S is the average number of sentences per 100 words in the text.
